runtime/runtime/src/actions.rs

```diff
@@ -99,21 +99,30 @@ pub(crate) fn execute_function_call(
         output_data_receivers,
     };
 
-    // TODO (#5920): enable chunk caching in the protocol. Also consider using RAII for switching the state back
-    // runtime_ext.set_trie_cache_mode(TrieCacheMode::CachingChunk);
-    let result = near_vm_runner::run(
-        &code,
-        &function_call.method_name,
-        runtime_ext,
-        context,
-        &config.wasm_config,
-        &config.transaction_costs,
-        promise_results,
-        apply_state.current_protocol_version,
-        apply_state.cache.as_deref(),
-    );
-    // runtime_ext.set_trie_cache_mode(TrieCacheMode::CachingShard);
-    result
+    // TODO (#5920): Consider using RAII for switching the state back
+    let protocol_version = runtime_ext.protocol_version();
+    let runner = |runtime_ext| {
+        near_vm_runner::run(
+            &code,
+            &function_call.method_name,
+            runtime_ext,
+            context,
+            &config.wasm_config,
+            &config.transaction_costs,
+            promise_results,
+            apply_state.current_protocol_version,
+            apply_state.cache.as_deref(),
+        )
+    };
+
+    if checked_feature!("protocol_feature_chunk_nodes_cache", ChunkNodesCache, protocol_version) {
+        runtime_ext.set_trie_cache_mode(TrieCacheMode::CachingChunk);
```

Comment:
Now I'm confused, why do you need to explicitly restrict the use of cache chunk to function calls?

Reply:
This is because we charge for touching trie nodes only during function calls.
I would use the cache for **all** trie accesses, but then I couldn't control chunk cache size, e.g. in case of multiple `get_account` calls.